adoc/deployment-bootstrap.adoc

```diff
@@ -290,6 +290,107 @@ these names will be used by the `AWS` cloud controller manager to reconcile node
+====
+
+===== Example Azure Cloud Provider Configuration
+
+====
+    {
+        "cloud":"AzurePublicCloud",
+        "tenantId": "<AZ_TENANT_ID>", // <1>
+        "aadClientId": "<AZ_AAD_CLIENT_ID>", // <2>
+        "aadClientSecret": "<AZ_AAD_CLIENT_SECRET>", // <3>
+        "subscriptionId": "<AZ_SUBSCRIPTION_ID>", // <4>
+        "resourceGroup": "<AZ_RESOURCE_GROUP>", // <5>
+        "location": "<AZ_LOCATION>", // <6>
+        "routeTableName": "<AZ_ROUTE_TABLE_NAME>", // <7>
+        "routeTableResourceGroup": "<AZ_ROUTE_TABLE_RESOURCE_GROUP>", // <8>
+        "useManagedIdentityExtension": true, // <9>
+        "useInstanceMetadata": true
+    }
+====
+<1> (required) The AAD Tenant ID for the Subscription that the cluster is deployed in.
+<2> (optional) The ClientID for an AAD application with RBAC access to talk to Azure RM APIs. This is used for service principle authentication.
+<3> (optional) The ClientSecret for an AAD application with RBAC access to talk to Azure RM APIs. This is used for service principle client secret authentication.
+<4> (required) The ID of the Azure Subscription that the cluster is deployed in.
+<5> (required) The name of the resource group that the cluster is deployed in.
+<6> (required) The location of the resource group that the cluster is deployed in.
+<7> (required) The name of the route table attached to the subnet that the cluster is deployed in.
+<8> (optional) The resource group name for routeTable. Default same as resourceGroup.
+<9> (optional) Use managed service identity for the virtual machine to access Azure ARM APIs. Set to false if not using managed identity authentication.
+
+After setting options in the `azure.conf` file, please proceed with <<cluster.bootstrap>>.
+
+[NOTE]
+====
+Clusters provisioned following link:{docurl}single-html/caasp-deployment/#_using_terraform[Deploying VMs from the Template] with `cpi_enable = true` automatically configurs Azure virtual machine to use a system-assigned managed identity.
+
+To use service principle authentication you need to <<cluster.bootstrap.azure.create-service-principle>> and provide `aadClientId`, `aadClientSecret` and configure `useManagedIdentityExtension` to `false`.
+====
+
+[IMPORTANT]
+====
+When cloud provider integration is enabled, it's very important to bootstrap and join nodes with the node names same as ~Azure~ virtual machine's hostnames.
+These names will be used by the `Azure` cloud controller manager to reconcile node metadata.
+====
+
+[[cluster.bootstrap.azure.create-service-principle]]
+===== Create Service Principle
+[IMPORTANT]
+====
+Ensure there is sufficient azure account permission. The account needs to be either a member of the `Owner` role, or have `Contributor` plus `User Access Administrator` roles.
+====
+
+. Install Azure CLI on local machine.
++
+For installation instructions, refer to link:https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest[Install the Azure CLI].
```

Comment:
We have `zypper install azure-cli` ^^

Reply:
cc @cclhsu

Reply:
Link includes instructions how to install with zypper.